x/participation/keeper/msg_withdraw_allocations_test.go

```diff
@@ -0,0 +1,145 @@
+package keeper_test
+
+import (
+	"testing"
+	"time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/stretchr/testify/require"
+
+	testkeeper "github.com/tendermint/spn/testutil/keeper"
+	"github.com/tendermint/spn/testutil/sample"
+	"github.com/tendermint/spn/x/participation/types"
+)
+
+func Test_msgServer_WithdrawAllocations(t *testing.T) {
+	var (
+		sdkCtx, tk, ts      = testkeeper.NewTestSetup(t)
+		ctx                 = sdk.WrapSDKContext(sdkCtx)
+		auctioneer          = sample.Address()
+		validParticipant    = sample.Address()
+		invalidParticipant  = sample.Address()
+		auctionSellingCoin  = sample.Coin()
+		auctionStartTime    = sdkCtx.BlockTime().Add(time.Hour)
+		validWithdrawalTime = auctionStartTime.Add(time.Hour * 24 * 30)
+	)
+
+	params := types.DefaultParams()
+	params.AllocationPrice = types.AllocationPrice{Bonded: sdk.NewInt(100)}
+	tk.ParticipationKeeper.SetParams(sdkCtx, params)
+
+	// delegate some coins so participant has some allocations to use
+	tk.DelegateN(sdkCtx, validParticipant, 100, 10)
+
+	// initialize an auction
+	tk.Mint(sdkCtx, auctioneer, sdk.NewCoins(auctionSellingCoin))
+	res, err := tk.FundraisingKeeper.CreateFixedPriceAuction(sdkCtx, sample.MsgCreateFixedAuction(
+		auctioneer,
+		auctionSellingCoin,
+		auctionStartTime,
+	))
+	require.NoError(t, err)
+	require.NotNil(t, res)
+	require.NotNil(t, res.BaseAuction)
+	auctionID := res.BaseAuction.Id
+
+	// validParticipant participates to auction
+	_, err = ts.ParticipationSrv.Participate(ctx, &types.MsgParticipate{
+		Participant: validParticipant,
+		AuctionID:   auctionID,
+		TierID:      1,
+	})
+	require.NoError(t, err)
+
+	preUsedAllocations, found := tk.ParticipationKeeper.GetUsedAllocations(sdkCtx, validParticipant)
+	require.True(t, found)
+
+	auctionUsedAllocations, found := tk.ParticipationKeeper.GetAuctionUsedAllocations(sdkCtx, validParticipant, auctionID)
+	require.True(t, found)
+	require.False(t, auctionUsedAllocations.Withdrawn)
```

Comment:
I think definition of variables read in the test body should be in the test body